src/uu/dd/src/parseargs/unit_tests.rs

```diff
@@ -0,0 +1,565 @@
+        String::from("if=foo.file"),
+        String::from("of=bar.file"),
+        String::from("ibs=10"),
+        String::from("obs=10"),
+        String::from("cbs=1"),
+        String::from("bs=100"),
+        String::from("count=2"),
+        String::from("skip=2"),
+        String::from("seek=2"),
+        String::from("status=progress"),
+        String::from("conv=ascii,ucase"),
+        String::from("iflag=count_bytes,skip_bytes"),
+        String::from("oflag=append,seek_bytes"),
+    ];
+    let args = args
+        .into_iter()
+        .fold(Vec::new(), append_dashes_if_not_present);
+
+    let matches = uu_app().get_matches_from_safe(args).unwrap();
+
+    assert_eq!(100, parse_ibs(&matches).unwrap());
+    assert_eq!(100, parse_obs(&matches).unwrap());
+    assert_eq!(1, parse_cbs(&matches).unwrap().unwrap());
+    assert_eq!(
+        CountType::Bytes(2),
+        parse_count(
+            &IFlags {
+                count_bytes: true,
+                ..IFlags::default()
+            },
+            &matches
+        )
+        .unwrap()
+        .unwrap()
+    );
+    assert_eq!(
+        200,
+        parse_skip_amt(&100, &IFlags::default(), &matches)
+            .unwrap()
+            .unwrap()
+    );
+    assert_eq!(
+        200,
+        parse_seek_amt(&100, &OFlags::default(), &matches)
+            .unwrap()
+            .unwrap()
+    );
+    assert_eq!(
+        StatusLevel::Progress,
+        parse_status_level(&matches).unwrap().unwrap()
+    );
+    assert_eq!(
+        IConvFlags {
+            ctable: Some(&EBCDIC_TO_ASCII_LCASE_TO_UCASE),
+            ..IConvFlags::default()
+        },
+        parse_conv_flag_input(&matches).unwrap()
+    );
+    assert_eq!(
+        OConvFlags::default(),
+        parse_conv_flag_output(&matches).unwrap()
+    );
+    assert_eq!(
+        IFlags {
+            count_bytes: true,
+            skip_bytes: true,
+            ..IFlags::default()
+        },
+        parse_iflags(&matches).unwrap()
+    );
+    assert_eq!(
+        OFlags {
+            append: true,
+            seek_bytes: true,
+            ..OFlags::default()
+        },
+        parse_oflags(&matches).unwrap()
+    );
+}
+
+#[ignore]
+#[test]
+// TODO: This should work, but Clap doesn't seem to understand it. Leaving it for now since the traditional dd if=foo.file works just fine.
+fn test_all_top_level_args_leading_dashes_sep_by_spaces() {
+    let args = vec![
+        String::from("dd"),
+        String::from("--if foo.file"),
+        String::from("--of bar.file"),
+        String::from("--ibs 10"),
+        String::from("--obs 10"),
+        String::from("--cbs 1"),
+        String::from("--bs 100"),
+        String::from("--count 2"),
+        String::from("--skip 2"),
+        String::from("--seek 2"),
+        String::from("--status progress"),
+        String::from("--conv ascii,ucase"),
+        String::from("--iflag count_bytes,skip_bytes"),
+        String::from("--oflag append,seek_bytes"),
```

Comment:
This fails because args are split at whitespace when typing them in a terminal. You'll have to do this here too, e.g. not `"--if foo.file"` but `"--if"` and then `"foo.file"` as another string.

However, this will also not pass the test because of the logic in `append_dashes_if_not_present`. For example, `--if foo.file --of bar.file` will be turned into `--if --foo.file --of --bar.file`.

Reply:
Glad for your help on this one. You've definitely helped to clarify some things I was struggling with. 

As a side note, I was hoping to be able to support both `dd --if foo.file ...` and `dd if=foo.file ...` since it felt like Clap was going to do that easily, but I didn't anticipate the issues you mentioned [above](https://github.com/uutils/coreutils/pull/2474#discussion_r675782227). 

To resolve this and the issue above, I've set all args that take values to require equals. Both `dd --if=foo.file ...` and `dd if=foo.file ...` appear to work and provide sensible error messages. However, there is probably more work to be done here in the future to make it a really excellent solution.

Reply:
Yeah, it doesn't need to be perfect from the beginning. We can always improve it